chain/store/splitstore/splitstore.go

```diff
@@ -0,0 +1,414 @@
+package splitstore
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/bmatsuo/lmdb-go/lmdb"
+
+	blocks "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+	dstore "github.com/ipfs/go-datastore"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/store"
+	"github.com/filecoin-project/lotus/chain/types"
+	bstore "github.com/filecoin-project/lotus/lib/blockstore"
+)
+
+const CompactionThreshold = 5 * build.Finality
```

Comment:
Naming nit: we should probably rename this and `compact` to `MigrationThreshold` and `migrate`, as that's effectively what we're doing.

Reply:
nah, I like compaction for this :)

Reply:
How is this a compaction and not a transfer/migration?

Reply:
Compaction has specific connotations in datastores, and this is certainly not a compaction. It's not about personal preference, it's about accuracy.

Reply:
Well, I am not going to argue names. It's not a simple migration however, it _does_ compact the datastore because it deletes unreachable nodes.

Reply:
Ok, we can revisit later. A rename would be cheap.